skyportal/models/user_notification.py

```diff
@@ -100,21 +101,29 @@ def user_preferences(target, notification_setting, resource_type):
         return
     if not target.user:
         return
-
-    if not target.user.preferences:
+    if not notification_setting:
         return
 
     if notification_setting == "email":
         if not email:
             return
         if not target.user.contact_email:
             return
-    elif notification_setting == "sms":
+    if notification_setting == "sms":
         if client is None:
             return
         if not target.user.contact_phone:
             return
-    elif notification_setting == "slack":
+
```

Comment:
Is there a reason this is stuck between sms and slack? Would maybe make more sense right after or even merged in with the `if notification_setting == "email"`.

Reply:
No you are right, I moved it up :) inside of the `if notification_setting == "email"`. However, the:
```
if not target.user.preferences:
        return
```
will be right between email, and SMS slack, simply because of that feature of "forced" notifications only on emails. Plus, for both slack and SMS, some things are needed in the preferences, down the line. For emails, it's simply about if it is activated in the preferences or not, whereas for the SMS and slack, some values in the preferences are needed in their respective notification methods (slack integration URL, SMS on shift or on a time slot...).

Reply:
That seems fine, as long as there's a logical reason to be stuck there.